Code/GraphMol/ChemTransforms/ChemTransforms.cpp

```diff
@@ -358,11 +375,45 @@ ROMol *replaceCore(const ROMol &mol, const ROMol &core,
       throw ValueErrorException(
           "Supplied MatchVect indices out of bounds of the target molecule");
     }
-
+    bool useMatch = false;
     if (replaceDummies || core.getAtomWithIdx(mvit.first)->getAtomicNum() > 0) {
       matchingIndices[mvit.second] = mvit.first;
+      useMatch = true;
     }
     allIndices[mvit.second] = mvit.first;
+    SideChainMapping mapping(mvit.second, mvit.first, useMatch);
+    matches.emplace_back(mvit.second, mapping);
+    if (molAtomsMapped[mvit.second]) {
+      multipleMappedMolAtoms.insert(mvit.second);
+    }
+    molAtomsMapped.set(mvit.second);
+  }
+
+  boost::dynamic_bitset<> multipleOwnedBonds(mol.getNumBonds());
+  if (multipleMappedMolAtoms.size() > 0) {
+    for (const auto &match : matches) {
+      const auto &mappingInfo = match.second;
+      if (multipleMappedMolAtoms.find(mappingInfo.molIndex) !=
+          multipleMappedMolAtoms.end()) {
+        auto coreAtom = core.getAtomWithIdx(mappingInfo.coreIndex);
+        CHECK_INVARIANT(
+            coreAtom->getDegree() == 1,
+            "Multiple core atoms match a mol atom, but one of the core "
+            "atoms has degree > 1 ");
+        auto coreNeighborIdx = *core.getAtomNeighbors(coreAtom).first;
+        auto molNeighborIdx =
+            std::find_if(matchV.cbegin(), matchV.cend(),
+                         [coreNeighborIdx](std::pair<int, int> p) {
+                           return p.first == static_cast<int>(coreNeighborIdx);
+                         })
+                ->second;
+        if (molNeighborIdx > -1) {
+          auto connectingBond =
```

Comment:
Since the next line could potentially seg fault, please add:
```
auto connectingBond =
    mol.getBondBetweenAtoms(mappingInfo.molIndex, molNeighborIdx);
CHECK_INVARIANT(connectingBond,"expected bond in molecule not found");
```

Reply:
Changed